client/src/components/productDetails/Product.jsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState, useEffect } from 'react';
+import axios from 'axios';
+import Image from './Image';
+import AddToCart from './AddToCart';
+import Style from './Style';
+
+function Product({ product }) {
+  const [styleList, setStyleList] = useState(null);
+  const [productObj, setProductObj] = useState(null);
+
+  // trying for one product for now
+  function getSpecificProduct() {
+    if (product) {
+      axios.get(`/products/${product.id}`, {
+      })
+        .then(({ data }) => {
+          console.log('Product info', data);
+          setProductObj(data);
+        })
+        .catch((err) => {
+          console.log('There was a problem in the server retrieving specific product data: ', err);
+        });
+    }
+  }
+
+  function getProductStyles() {
+    if (product) {
+      axios.get(`/products/${product.id}/styles`, {
+      })
+        .then(({ data }) => {
+          console.log('Product Styles available', data.results);
+          setStyleList(data.results);
+        })
+        .catch((err) => {
+          console.log('There was a problem in the server retrieving specific product styles: ', err);
+        });
+    }
+  }
+
+  useEffect(() => {
+    getSpecificProduct();
+    getProductStyles();
+  }, [product]);
```

Comment:
This looks good, but just a recommendation to look into axios.all and spread. It allows you to make multiple axios requests at once so you don't have to create two different functions like this!

Reply:
will look into all function!